packages/swapper/src/swappers/ZrxSwapper/getZrxTradeQuote/getZrxTradeQuote.ts

```diff
@@ -154,3 +169,196 @@ export async function getZrxTradeQuote(
     )
   }
 }
+
+async function _getZrxPermit2TradeQuote(
+  input: GetEvmTradeQuoteInput,
+  assertGetEvmChainAdapter: (chainId: ChainId) => EvmChainAdapter,
+  assetsById: AssetsByIdPartial,
+  zrxBaseUrl: string,
+): Promise<Result<TradeQuote, SwapErrorRight>> {
+  const {
+    sellAsset,
+    buyAsset,
+    accountNumber,
+    receiveAddress,
+    affiliateBps,
+    potentialAffiliateBps,
+    chainId,
+    supportsEIP1559,
+    sellAmountIncludingProtocolFeesCryptoBaseUnit,
+  } = input
+
+  const slippageTolerancePercentageDecimal =
+    input.slippageTolerancePercentageDecimal ??
+    getDefaultSlippageDecimalPercentageForSwapper(SwapperName.Zrx)
+
+  const sellAssetChainId = sellAsset.chainId
+  const buyAssetChainId = buyAsset.chainId
+
+  if (!isSupportedChainId(sellAssetChainId)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `unsupported chainId`,
+        code: TradeQuoteError.UnsupportedChain,
+        details: { chainId: sellAsset.chainId },
+      }),
+    )
+  }
+
+  if (!isSupportedChainId(buyAssetChainId)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `unsupported chainId`,
+        code: TradeQuoteError.UnsupportedChain,
+        details: { chainId: sellAsset.chainId },
+      }),
+    )
+  }
+
+  if (sellAssetChainId !== buyAssetChainId) {
+    return Err(
+      makeSwapErrorRight({
+        message: `cross-chain not supported - both assets must be on chainId ${sellAsset.chainId}`,
+        code: TradeQuoteError.CrossChainNotSupported,
+        details: { buyAsset, sellAsset },
+      }),
+    )
+  }
+
+  const maybeZrxPriceResponse = await fetchFromZrxPermit2({
```

Comment:
```suggestion
  const maybeZrxQuoteResponse = await fetchFromZrxPermit2({
```

See https://github.com/shapeshift/web/pull/7758#discussion_r1806278116 for reference

Reply:
Per reply above, we can come back and simplify everything when removing the V1 api.